app/src/main/java/me/hufman/androidautoidrive/carapp/music/views/PlaybackView.kt

```diff
@@ -368,6 +375,8 @@ class PlaybackView(val state: RHMIState, val controller: MusicController, val ca
 		if (displayedSong != controller.getMetadata() ||
 				displayedConnected != controller.isConnected()) {
 			redrawSong()
+		} else {
+			redrawLongTitles()
```

Comment:
how often does `redraw()` get invoked? Is it a tick based system? Just curious if it's worth the effort to avoid redrawing if the text values don't change - which really depends on how often we redraw I guess

Reply:
The music app redraws at 2hz, to update when the progress bar moves. Most of the Playback View has its own caching (displayedSong and so on), but the RHMIApplicationIdempotent layer handles avoiding any unnecessary data updates to the car :) RHMIApplicationIdempotent suppresses any updates with identical strings or numbers, but displayedSong avoids recompressing coverart and so on.